worker/docker_image_manager.py

```diff
@@ -0,0 +1,140 @@
+import threading
+import os
+import time
+import traceback
+import logging
+import json
+
+import file_util
+from formatting import size_str
+
+logger = logging.getLogger(__name__)
+
+
+class DockerImageManager(object):
+    """
+    Manages the Docker images available on the worker, and ensures that they
+    do not fill up the disk.
+
+    The last-used dates are only tracked for images on which
+    `DockerImageManager.touch_image()` has been called. All other images will
+    be left untouched by the cleanup thread.
+
+    DependencyManager should be instantiated before DockerImageManager, to
+    ensure that the work directory already exists.
+    """
+    STATE_FILENAME = 'images-state.json'
+
+    def __init__(self, docker, work_dir, min_free_bytes=1e9,
```

Comment:
The problem being that the actual allowed size of the image cache may effectively be smaller, if there are other files on the same disk besides bundles and images.